src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -1305,6 +1305,28 @@ public GraphNode GetFirstDirtyGraphNode(int pc, int classIndex, int procIndex)
             return null;
         }
 
+        /// <summary>
+        /// Gets the first dirty graphnode
+        /// </summary>
+        /// <param name="pc"></param>
+        /// <param name="classIndex"></param>
+        /// <param name="procIndex"></param>
+        /// <returns></returns>
+        public GraphNode GetFirstDirtyGraphNode(int classIndex, int procIndex)
+        {
+            List<GraphNode> gnodeList = GetGraphNodesAtScope(classIndex, procIndex);
+            if (gnodeList != null && gnodeList.Count > 0)
```

Comment:
```
if (gNodeList != null)
{
    return gnodeList.FirstOrDefault(node => node.isActive && node.isDirty);
}
```
BTW, we may change the return value of ``GetGraphNodesAtScope()`` so that it always returns non-null value.